calico_containers/pycalico/netns.py

```diff
@@ -180,6 +179,32 @@ def remove_ip_from_ns_veth(container_pid, ip, veth_name_ns):
                        "dev", "%(device)s" % veth_name_ns])
 
 
+def provision_veth(self, endpoint, ns_pid, veth_name_ns):
+    """
+    Create the veth, move into the container namespace, add the IP and
+    set up the default routes.
+
+    Note, the endpoint will not be updated in ETCD. If desired, the user
+    should update the endpoint mac with the mac address provided
+    by the function and then call update_endpoint
+
+    :param endpoint: The endpoint object to provision the veth on
+    :param ns_pid: The PID of the namespace to operate in
+    :param veth_name_ns: The name of the interface in the namespace
+    :return The mac address of the veth as a string
+    """
+    create_veth(endpoint.name, endpoint.temp_interface_name)
+    for ip in endpoint.ipv4_nets | endpoint.ipv6_nets:
```

Comment:
Putting a little more meat on the bone, ipv*X*_nets are, for now, just the IP addresses, but encoded as networks for future-proofing the data model.

In particular, IPv6 SLAAC assigns a /64 to each host, allowing for multiple interface IDs without changing the routing behavior.